rfc/rfc-58/rfc-58.md

```diff
@@ -0,0 +1,69 @@
+<!--
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+# RFC-58: Integrate column stats index with all query engines
+
+
+
+## Proposers
+
+- @pratyakshsharma
+
+## Approvers
```

Comment:
@alexeykudinkin and @prasannarajaperumal would be good reviewers for this IMO. @bhasudha if you don't mind, they can run point on this?

Reply:
+1 - Yes I will take over this. @pratyakshsharma I will setup some time to sync up once I am up to speed on the work done in the PR. Overall my thoughts so far
1. Rebuilding a generic SQL expression tree and implementing the entire pruning logic inside Hudi - I am not a big fan of this idea. Each engine might have multiple pruning passes after certain optimization and do expression pruning differently. It is hard to generalize this model accross engines. 
2. We have to implement a way to 
a. Prune the files while listing based on physical columns used in expressions (similar to ColumnHandle approach). Ideally we represented the pruned list as a bitset - as typically pruning goes through multiple passes and passing just the bitset along to get the final list of files to be scanned. 
b. We need to propagate the column min/max/null and other stats along with the return API (not just FileStatus[]) for the engines to further do pruning on the expressions involving these columns. This will enable optimizations like join elimination, false filter folding etc. 
I am not sure if Presto/Flink/Spark has clean API abstraction to do this.

I will chat with you to understand and explain this a bit more. 

cc @vinothchandar

Reply:
@prasannarajaperumal thank you for your inputs. Sure, lets discuss more once you have gone through the changes. We are already connected on Hudi slack.

Reply:
@prasannarajaperumal please add me to this forum as well

Reply:
@prasannarajaperumal I have elaborated the ColumnHandle/ColumnDomain approach in a bit more detail. Sorry for the long hold on this one. Please have a look.